### DISCUSSION TITLE
[QUESTION] Does SDB comply to GDB's machine interface?

### DISCUSSION THREAD

**Discussion by tibel** [mono/sdb#81] [Questions] [answered]
I don't know if this is the right place to ask this question, but I could not find any info on this topic.

Does SDB comply to GDB's machine interface?

I ask because of [Microsoft/MIEngine](https://github.com/Microsoft/MIEngine) and remote debugging a mono project from inside Visual Studio?

**Comment by vargaz** [comment, marked as answer]
It doesn't, it uses a custom wire protocol based on the java debugger protocol:
http://www.mono-project.com/docs/advanced/runtime/docs/soft-debugger/

**Comment by tibel** [comment]
Please correct me where I'm wrong:
SDB talks to the mono runtime using a custom wire protocol (what you described).
GDB mi interface is ASCII based and defines the commands and output on stdin and stdout.
So theoretically when commands and output are compatible to gdb mi, this could work.

VS gdb support:
- https://blogs.msdn.microsoft.com/vcblog/2015/11/18/announcing-the-vs-gdb-debugger-extension/
- https://channel9.msdn.com/Events/Visual-Studio/Connect-event-2015/807

**Comment by alexrp** [comment]
SDB (the command line program) doesn't support the machine interface out of the box as Zoltan said, but I imagine it would be possible to support it in theory, as long as it's just a matter of supporting certain commands.

**Comment by alexrp** [comment]
Closing as the question has been answered.
